app/assets/javascripts/mobile/mobile_file_uploader.js

```diff
@@ -4,78 +4,91 @@
 function createUploader(){
    var aspectIds = gon.preloads.aspect_ids;
 
-   new qq.FileUploaderBasic({
-       element: document.getElementById('file-upload-publisher'),
-       params: {'photo' : {'pending' : 'true', 'aspect_ids' : aspectIds},},
-       allowedExtensions: ['jpg', 'jpeg', 'png', 'gif', 'tiff'],
-       action: "/photos",
-       debug: false,
-       button: document.getElementById('file-upload-publisher'),
-       sizeLimit: 4194304,
+  new qq.FineUploaderBasic({
+    element: document.getElementById("file-upload-publisher"),
+    request: {
+      endpoint: Routes.photos(),
+      params: {
+        /* eslint-disable camelcase */
+        authenticity_token: $("meta[name='csrf-token']").attr("content"),
+        photo: {
+          aspect_ids: aspectIds,
+          /* eslint-enable camelcase */
+          pending: true
+        }
+      },
+      paramsInBody: false
+    },
+    validation: {
+      allowedExtensions: ["jpg", "jpeg", "png", "gif", "tif", "tiff"],
+      sizeLimit: 4194304
+    },
+    debug: false,
+    button: document.getElementById("file-upload-publisher"),
 
-       onProgress: function(id, fileName, loaded, total){
-        var progress = Math.round(loaded / total * 100 );
-         $('#fileInfo-publisher').text(fileName + ' ' + progress + '%');
-       },
-
-       messages: {
-          typeError: Diaspora.I18n.t("photo_uploader.invalid_ext"),
-          sizeError: Diaspora.I18n.t("photo_uploader.size_error"),
-          emptyError: Diaspora.I18n.t("photo_uploader.empty")
-       },
-
-       onSubmit: function(){
-        $('#file-upload-publisher').addClass("loading");
-        $('#publisher_textarea_wrapper').addClass("with_attachments");
-        $('#photodropzone').append(
+    callbacks: {
+      onProgress: function(id, fileName, loaded, total) {
+        var progress = Math.round(loaded / total * 100);
+        $("#fileInfo-publisher").text(fileName + " " + progress + "%");
+      },
+      onSubmit: function() {
+        $("#file-upload-publisher").addClass("loading");
+        $("#publisher_textarea_wrapper").addClass("with_attachments");
+        $("#photodropzone").append(
           "<li class='publisher_photo loading' style='position:relative;'>" +
-            "<img alt='Ajax-loader2' src='"+ImagePaths.get('ajax-loader2.gif')+"' />" +
+          "<img alt='Ajax-loader2' src='" + ImagePaths.get("ajax-loader2.gif") + "' />" +
           "</li>"
-          );
-       },
-
-       onComplete: function(_id, fileName, responseJSON) {
-        $('#fileInfo-publisher').text(Diaspora.I18n.t("photo_uploader.completed", {'file': fileName}));
+        );
+      },
+      onComplete: function(_id, fileName, responseJSON) {
+        $("#fileInfo-publisher").text(Diaspora.I18n.t("photo_uploader.completed", {"file": fileName}));
         var id = responseJSON.data.photo.id,
```

Comment:
Again, this will also be called when the upload failed, so you have to make sure, that you only do this kind of stuff when the upload was successful.

(optional)